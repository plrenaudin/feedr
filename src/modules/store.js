import { observable, action, toJS, configure } from "mobx";
import db, { EMPTY_RECORD } from "./db";
import { shortFormatDate } from "./formatter";
import addDays from "date-fns/add_days";

configure({ enforceActions: true });

export default class Store {
  day = observable({ ...EMPTY_RECORD, date: shortFormatDate() });
  constructor() {
    this.load(shortFormatDate());
  }

  set = action(({ date = shortFormatDate(), morning = [], noon = [], evening = [] }) => {
    this.day.date = date;
    this.day.morning = morning;
    this.day.noon = noon;
    this.day.evening = evening;
  });

  addItem = action((meal, data) => {
    this.day[meal].push(data);
    this.save();
  });

  removeItem = action((meal, index) => {
    this.day[meal].splice(index, 1);
    this.save();
  });

  nextDay() {
    this.load(shortFormatDate(addDays(this.day.date, 1)));
  }

  previousDay() {
    this.load(shortFormatDate(addDays(this.day.date, -1)));
  }

  load = action(date => {
    return db.get(date).then(data => {
      this.set(data || { date });
    });
  });

  save() {
    return db.set(this.day.date, toJS(this.day));
  }
}
