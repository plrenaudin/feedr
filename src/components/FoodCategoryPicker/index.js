import { Component, linkEvent } from "inferno";
import { inject } from "inferno-mobx";
import { FOOD_TYPES } from "../../modules/store";
import Icon from "../Icon";

class FoodCategoryPicker extends Component {
  constructor(props, context) {
    super(props, context);
    this.meal = props.meal;
    this.mealItemIndex = props.index;
    this.store = props.store;
    this.state = {
      opened: false,
      selected: this.store.day[this.meal][this.mealItemIndex].categories || []
    };
  }

  openPicker(instance) {
    instance.setState({ opened: true });
    instance.context.isSwipable = false;
  }

  closePicker(instance) {
    instance.setState({ opened: false });
    instance.context.isSwipable = true;
    instance.store.editItemCategory(instance.meal, instance.mealItemIndex, instance.state.selected);
  }

  toggleItem(key) {
    return instance => {
      const idx = instance.state.selected.indexOf(key);
      if (idx > -1) {
        const copy = instance.state.selected.slice();
        copy.splice(idx, 1);
        instance.setState({ selected: copy });
      } else {
        instance.setState({ selected: instance.state.selected.concat(key) });
      }
    };
  }

  render() {
    return (
      <div class="food-type-picker">
        <a role="button" onClick={linkEvent(this, this.openPicker)}>
          <Icon name="price-tag" />
        </a>
        {this.state.opened && (
          <div class="picker">
            <div class="picker-overlay" onClick={linkEvent(this, this.closePicker)} />
            <div class="picker-content">
              <ul $HasKeyedChildren>
                {Object.keys(FOOD_TYPES).map(key => (
                  <li
                    key={key}
                    class={`color${key} ${this.state.selected.includes(key) && "selected"}`}
                    onClick={linkEvent(this, this.toggleItem(key))}
                  >
                    {FOOD_TYPES[key]}
                    <span class="check">
                      <Icon name="checkmark" />
                    </span>
                  </li>
                ))}
              </ul>
            </div>
          </div>
        )}
      </div>
    );
  }
}

export default inject("store")(FoodCategoryPicker);
