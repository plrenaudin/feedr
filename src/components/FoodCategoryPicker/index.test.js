import { render } from "inferno";
import FoodCategoryPicker from ".";
import store from "../../../test/fixture/store";
import { Provider } from "inferno-mobx";

it("Matches the snapshot", () => {
  const div = document.createElement("div");
  expect(
    render(
      <Provider store={store}>
        <FoodCategoryPicker meal="morning" index={0} />
      </Provider>,
      div
    ).props.children.dom
  ).toMatchSnapshot();
});
