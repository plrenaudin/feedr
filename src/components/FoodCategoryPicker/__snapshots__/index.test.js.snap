// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<div
  class="food-type-picker"
>
  <a
    role="button"
  >
    <span
      class="icon-wrapperundefined"
    >
      <svg
        class="icon"
      >
        <use
          xlink:href="#icon-price-tag"
        />
      </svg>
    </span>
  </a>
</div>
`;
